### ISSUE TITLE
Yaml subresource error

### ISSUE THREAD

**Issue by cacahouwete** [api-platform/core#3110]
| label | version |
| --- | --- |
| api-platform/core | 2.4.5 |
| php | 7.2.21 |
| symfony/* | 4.3.3 |

Hello, everybody.

I have a little problem when I configure a subresource with Yaml file configuration but if I put the same configuration in XML file it works. My files:

```yaml
App\Domain\TourBuilding\Model\AggregateRoot\Plan\Plan:
  properties:
    resources:
      subresource: { collection: true, resourceClass: 'App\Domain\TourBuilding\Model\Aggregate\Resource\Resource' }
```
```xml
<?xml version="1.0" ?>

<resources xmlns="https://api-platform.com/schema/metadata"
           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
           xsi:schemaLocation="https://api-platform.com/schema/metadata
           https://api-platform.com/schema/metadata/metadata-2.0.xsd">
    <resource
            class="App\Domain\TourBuilding\Model\AggregateRoot\Plan\Plan"
    >
        <property name="resources">
            <subresource collection="true" resourceClass="App\Domain\TourBuilding\Model\Aggregate\Resource\Resource" />
        </property>
    </resource>
</resources>
```

The error:
```shell
In SubresourceMetadata.php line 27:
                                                                               
  Argument 1 passed to ApiPlatform\Core\Metadata\Property\SubresourceMetadata  
  ::__construct() must be of the type string, null given, called in /app/vend  
  or/api-platform/core/src/Metadata/Property/Factory/ExtractorPropertyMetadat  
  aFactory.php on line 150                                                     
```

**Comment by soyuka** [comment]
We have the exact same configuration in our tests: https://github.com/api-platform/core/blob/22630f78f09179237d8d5535c528dda5d3f80b99/tests/Fixtures/FileConfigurations/resources.yml#L35 

Maybe because your property is called `resources`?

**Comment by temp** [comment]
We just stumbled upon the same error, and we did some research - in our case it's code like this that produces the error:

```
class Foo {
  /** @var Bar[] */
  private array $bars;
}
class Bar {
}
```

As sson as we remove the array type hint, it works.
We digged into the code, and it's related to the ReflectionExtractor, which extract an array type, with a class string.
When we remove the type hint, type is completely empty and another mechanism kicks in I guess.
